components/table/Column.razor.cs

```diff
@@ -214,20 +310,50 @@ private void FilterSelected(TableFilter<TData> filter)
             StateHasChanged();
         }
 
-        private void FilterConfirm()
+        private void FilterConfirm(bool isReset = false)
         {
             _filterOpened = false;
+            if (!isReset && _columnFilterType == TableFilterType.FeildType) Filters?.ForEach(f => { if (!f.Selected && f.Value != null) f.Selected = true; });
```

Comment:
When `TData` is value type, `f.Value != null` will always be `true`, unless user sets `TData` to `Nullable<T>`, like `DateTime?`.

Reply:
If user want to get null value row, it won't work.
Suggestion: add checkbox to control filter option is selected or not, rather than check the filter option value is null or not null.